### PR TITLE
mgr/dashboard: Store user table configurations

### PR DESCRIPTION
**Pull request by Devp00l**
The feature stores sorting, displayed columns and the table limit for
each table, that does not forbid the auto saving feature.

Signed-off-by: Stephan Müller <smueller@suse.com>

This PR is depends on [#20807](https://github.com/ceph/ceph/pull/20807).

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-user-config.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { SortPropDir } from '@swimlane/ngx-datatable';
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-user-config.ts]: Please add an empty line below.

Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-user-config.ts]: done

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by Devp00l** [comment]
jenkins retest this please

**Comment by tspmelo** [comment]
The use of Proxy is causing problems when running unit tests with PhantomJS.
You will need to find a working polyfill or change your code.

**Comment by rjfd** [comment]
@Devp00l please rebase

**Comment by Devp00l** [comment]
I rebased the PR on the current master and found a work around for the phantomJS limitations.

**Comment by votdev** [comment]
Is it possible to introduce a unit test helper that wraps the PhantomJS things? Many tests containing the same duplicate code now.

**Comment by tspmelo** [comment]
With #22031 we can test this PR on the CLI without the need for changing the tests code.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts [hunk 2]
```diff
@@ -168,9 +175,87 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
...
+      set(config, prop, value) {
+        config[prop] = value;
+        observer.next(config);
+        return true;
+      }
+    });
+  }
+
+  _saveUserConfig (config) {
+    this.localStorage[this.tableName] = JSON.stringify(config);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts]: Please replace this with `this.localStorage.setItem(this.tableName, JSON.stringify(config))`. 
This change is needed for Jest.

**Pull request merged**